packages/esm-ward-app/src/ward-view/ward-view.resource.ts

```diff
@@ -46,14 +47,25 @@ export function filterBeds(admissionLocation: AdmissionLocationFetchResponse): B
   return bedLayouts;
 }
 
-export function getWardMetrics(bedLayouts: BedLayout[], wardPatientGroup: WardPatientGroupDetails): WardMetrics {
-  const patients = wardPatientGroup?.totalPatientsCount ?? 0;
-  const totalBeds = bedLayouts?.length ?? 0;
-  const occupiedBeds = bedLayouts?.filter((bed) => bed.patients?.length > 0).length ?? 0;
+export function getWardMetrics(wardPatientGroup: WardPatientGroupDetails): WardMetrics {
+  // pull all the patients out of the three constructs they are stored in: unadmitted but in a bed, admitted and in a bed, and admitted but not in a bed
+  const allPatients = [
+    ...wardPatientGroup.wardUnadmittedPatientsWithBed?.values(),
+    ...[...wardPatientGroup.wardAdmittedPatientsWithBed?.values()].map((admission) => admission.patient),
+    ...wardPatientGroup.wardUnassignedPatientsList?.map((admission) => admission.patient),
+  ];
+
+  const patientCount = allPatients?.length ?? 0;
+  const newborns = filterNewborns(allPatients)?.length ?? 0;
+  const femalesOfReproductiveAge = filterReproductiveAge(filterFemale(allPatients))?.length ?? 0;
```

Comment:
Basically, we are calculating the "maternal" metrics in all cases, but the individual component is configured to display them or not.